FunMotifs.pm

```diff
@@ -0,0 +1,200 @@
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 CONTACT
+
+ Ensembl <http://www.ensembl.org/info/about/contact/index.html>
+
+=cut
+
+=head1 NAME
+
+ FunMotifs
+
+=head1 SYNOPSIS
+
+ mv FunMotifs.pm ~/.vep/Plugins
+ ./vep -i variations.vcf --plugin FunMotifs,/path/to/funmotifs/all_tissues.bed.gz,all,uterus
+ ./vep -i variations.vcf --plugin FunMotifs,/path/to/funmotifs/blood.funmotifs_sorted.bed.gz,individual,fscore,dnase_seq
+ 
+ Parameters Required:
+ 
+ [0] : FunMotifs BED file
+ [1] : Type of file provided ('all' or 'individual')
+ [2]+ : List of columns to include within VEP output (e.g. fscore, skin, contactingdomain)
+ 
+
+=head1 DESCRIPTION
+
+ This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
+ adds tissue-specific transcription factor motifs from FunMotifs to VEP output.
+
+ Please cite the FunMotifs publication alongside the VEP if you use this resource, which you can find on:
+ http://bioinf.icm.uu.se:3838/funmotifs/
+ 
+ FunMotifs files can be downloaded from: http://bioinf.icm.uu.se:3838/funmotifs/
+
+ The tabix utility must be installed in your path to use this plugin.
+
+=cut
+package FunMotifs;
+
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin;
+
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin);
+
+
+sub new {
+  my $class = shift;
+
+  my $self = $class->SUPER::new(@_);
+  if(scalar(@{$self->params}) < 3){
+    warn('Insufficient input parameters found for FunMotifs plugin');
+    return $self;
+  }
+  $self->expand_left(0);
+  $self->expand_right(0);
+
+  $self->get_user_params();
+
+  my $file = $self->params->[0];
+
+  return $self;
+}
+
+sub feature_types {
+  return ['Transcript'];
+}
+
+sub get_header_info {
+  return { FunMotifs => 'Annotated Transcription Factor Motifs'};
+}
+
+sub run {
+  my ($self, $tva) = @_;
+
+  my $filetype = 'individual';
+  my $params = $self->params;
+  
+  if(scalar(@$params) > 2 && $params->[1] eq 'all'){
+    $filetype = 'all';
+  }
+  
+  my $output_prefix = 'FM';
+  
+  $self->{motif_filetype} = $filetype;
+  
+  my $tissue_type = $params->[1];
+
+  my $vf = $tva->variation_feature;
+
+  my $column_headers = {
+                         all => [qw(name score pval strand blood brain breast cervix colon esophagus kidney liver lung myeloid pancreas prostate skin stomach uterus)],
+                         individual => [qw(name score pval strand fscore chromhmm contactingdomain dnase_seq fantom loopdomain numothertfbinding othertfbinding replidomain tfbinding tfexpr)],
+                       };
+
+  my ($res) = @{$self->get_data($vf->{chr}, $vf->{start}, $vf->{end})};
+
+  shift $params;
```

Comment:
Can we add a comment about which parameter is removed from the list ?
Should we do the same as GeneSplicer ?
https://github.com/Ensembl/VEP_plugins/blob/078cf3baec7c0413b3fc64656daa4e7d50140a3c/GeneSplicer.pm#L118-L120

Reply:
There's an earlier check on parameter existence, so we don't need to add in warning messages, but I've added code comments to make it clearer